### PR TITLE
Add pylint, pycodestyle and mypy check to CircleCI

### PR DESCRIPTION
**Pull request by marmistrz**

### CODE REVIEW

File: lintdiff.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+"$@" > $REF_OUT
+
+# Take back the checkout
+git reset --hard HEAD
+git checkout "$CURRENT_BRANCH" || exit 1
+
+
+diff=$(diff --old-line-format="" --unchanged-line-format="" -w <(sort $REF_OUT) <(sort $CURRENT_OUT))
+# There's always a newline, so -gt 1
+if [ "$diff" != "" ]; then
```
Comment (banasiakadam60) [inline lintdiff.sh]: You can use `-n` operator (non-empty)
``if [[ -n "${diff}" ]]; then`

**Review by banasiakadam60 — APPROVED** [review]

**Comment by marmistrz** [comment]
The `mypy` checks are broken ATM: https://github.com/python/mypy/issues/3850

**Comment by marmistrz** [comment]
Do not merge yet.

**Comment by marmistrz** [comment]
This will be ready to merge once https://github.com/golemfactory/golem/pull/1283 is merged.

**Review by banasiakadam60 — APPROVED** [review]

**Comment by banasiakadam60** [comment]
LGTM

**Pull request merged**